### ISSUE TITLE
[Bug] 

### ISSUE THREAD

**Issue by jjcv2304** [AzureAD/microsoft-identity-web#1423]
**Which version of Microsoft Identity Web are you using?**
Microsoft Identity Web 1.16.0

**Where is the issue?**
* Web app
    * [X] Sign-in users
    * [ ] Sign-in users and call web APIs
* Web API
    * [ ] Protected web APIs (validating tokens)
    * [ ] Protected web APIs (validating scopes)
    * [ ] Protected web APIs call downstream web APIs
* Token cache serialization
     * [ ] In-memory caches
     * [ ] Session caches
     * [ ] Distributed caches
* Other (please describe)

**Is this a new or an existing app?**
This is a prove of concept before we migrate from NET Framework to NET Core

**Repro**
I configure an Azure AD provider and an Azure B2C provider as explained in https://github.com/AzureAD/microsoft-identity-web/wiki/multiple-authentication-schemes (I also tried with the sample app from here https://github.com/AzureAD/microsoft-identity-web/tree/master/tests/MultipleAuthSchemes) 
When I Sign In using a user from the Azure AD tenant it works as expected but when I try to Sign In with a user from the Azure B2C tenant it doesn't work.

When using a B2C account(either internal or guest user) after entering the password I get an error that says something like your B2C user does not exist in the AD tenant (which should be ok since it's a B2C account and not an AD one) the message format is like this
 > AADSTS50020: User account 'xxx@B2CTenant.onmicrosoft.com' from identity provider xxx' does not exist in tenant ADTenant' and cannot access the application ADApplication in that tenant. The account needs to be added as an external user in the tenant first. Sign out and sign in again with a different Azure Active Directory user account.

**Expected behavior**
When I use an Azure AD account to Sign In it should be able to sign in (which already is working ok)
When I use an Azure B2C account(internal or guest) I should be able to sign in, without having a dependency on a different Azure Active Directory. In the future I would like to have a B2C provider + AAD Multitenant.

**Actual behavior**
When using an Azure B2C account I get an error message saying that the user is not authorized in another Azure Active Directory, something like this:
 > AADSTS50020: User account 'xxx@B2CTenant.onmicrosoft.com' from identity provider xxx' does not exist in tenant ADTenant' and cannot access the application ADApplication in that tenant. The account needs to be added as an external user in the tenant first. Sign out and sign in again with a different Azure Active Directory user account.

**Possible solution**

**Additional context / logs / screenshots**

**Comment by jmprieur** [comment]
@jjcv2304 did you use 2 buttons? one to sign-in to AAD, and one to sign-in to B2C? 
The error you get comes from AAD (AADSTS). Which authority did you use?
Did you see this sample? https://github.com/AzureAD/microsoft-identity-web/tree/master/tests/MultipleAuthSchemes

**Comment by jjcv2304** [comment]
Thanks @jmprieur. 

No, I didn't implement 2 buttons. I didn't got that part from reading the Readme.

I tried the sample application you mention, but the first thing I saw when launching the app, is a logging page and if I use an AAD account then it worked but if I used a B2C account it didn't...

After your comment I took a closer look at the sample app and understood a bit better how it works. Now I can launch the app, using `[AllowAnonymous]` in the Home\Index and then choose which Sign In method I want to use.

So thank you for your help!

PS: My next step will be to combine B2C and Multitenant AAD, so far the sample app you mention is the closer sample I have seen that may solve my requirements. Would it be correct to think that a combination of the two following sample apps is the path I should follow?:

- https://github.com/AzureAD/microsoft-identity-web/tree/master/tests/MultipleAuthSchemes

- https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/tree/master/2-WebApp-graph-user/2-3-Multi-Tenant
